docs/source/conf.py

```diff
@@ -402,50 +411,71 @@ def generate_weights_table(module, table_name, metrics, dataset, include_pattern
     generated_dir = Path("generated")
     generated_dir.mkdir(exist_ok=True)
     with open(generated_dir / f"{table_name}_table.rst", "w+") as table_file:
+        table_file.write(
+            f"Table of all available {table_name.replace('_',' ').title()} Weights \n{(32 + len(table_name))*title_character}\n"
+        )
+        table_file.write(f"{table_description}\n\n")
```

Comment:
IIUC this is the same as before but instead of writing these lines in the `.rst` file (as preferred), we're now generating it here and writing it in the table files.

Could you explain what the difference is, and why it "works"?

Reply:
So there was no title/headings in the generated table rst files, that is why we were getting results like below
 (notice the `<no title>`)

![image](https://user-images.githubusercontent.com/72816663/171637097-380e04fc-0845-498d-b039-31957b086b7c.png)

So to solve this, we needed to add the titles in the generated table files and not manually write them in `models.rst`. 
And the descriptions for the tables should be written after the headings. So there was no choice but to add the description in the generated table rst.

Reply:
> It feels like we're patching a limitation of sphinx's search by working around it without addressing the actual core of the issue. Did we figure out why writing the title within the file makes the search render better?

Yeah maybe! 

> which is better because we have the title, but it still looks broken.
Agree! Closing this pr as it doesn't properly solve the issue.

Reply:
Maybe someone with a good understanding of the sphinx theme  could have a look at this! 

maybe @ain-soph (sorry for shameless tagging, i really liked how you modified the theme for your project) If you have spare time ;-;